cue/ValidActions.cue

```diff
@@ -1,28 +1,38 @@
-#Steps: {
-    name?: string
-    "timeout-minutes": *10 | >=0
+#steps: "steps": {
+			"Steps1": {
+				"name":            "string"
+				"timeout-minutes": 10
+				"uses":            "actions/checkout@v2"
+				"with": {
+					"repo": "string"
+					"branch": "string"
+				}
+			}
+		}
 
-    *{
-        uses: "actions/checkout@v2"
-        with?: [string]: string
-    } | {
-        run: [...string]
-    }
-    ...
+name: "string"
+env: {
+	"ENVIR":  "ENV"
+	"ENVIR2": "ENV2"
+	"ENVIR3": "ENV3"
 }
-
-#job: {
-    name: string
-    "runs-on": "ubuntu-latest"
-    container?: {
-        image: string
-    }
-    needs?: [...string]
-    steps: [...#Steps]
-    ...
+on: {
+	"push-to": [
+		"string1",
+		"string2",
+	]
+}
+jobs: {
+	"job": {
+		"name":    "job1"
+		"runs-on": "ubuntu-latest"
+		"container": {
+			"image": "string"
+		}
+		"needs": [
+			"job1",
+			"job2",
+		]
+		#steps
+	}
```

Comment:
What is this? It's not a valid GitHub action schema

Reply:
Yes but we tried to retrieve any info from the original CUE program that validate github actions but it seems like it is impossible to get it if the fields are requirements and not values.

Reply:
![Screenshot from 2022-05-22 21-28-37](https://user-images.githubusercontent.com/71719097/169712878-c4a6d86e-ccf3-4daf-998b-ad6fc6aec06b.png)

Reply:
![Screenshot from 2022-05-22 21-28-37](https://user-images.githubusercontent.com/71719097/169712881-d4fe527f-726b-4351-8581-b3849cd36ff4.png)

Reply:
![Screenshot from 2022-05-22 21-26-39](https://user-images.githubusercontent.com/71719097/169712897-80d693e1-59bc-43a7-ac70-d991815fcc4c.png)

Reply:
![new-screen](https://user-images.githubusercontent.com/71719097/170710005-8b1302d6-75e6-4367-a415-9cdfa10b1e3c.png)

Reply:
Yeah it's normal, you must retrieve/read a file that implement the definition, not the definition itself haha